### PR TITLE
 Decommission old /leadership page template (Fixes #15135) 

### LINKED ISSUE #15135: Remove old leadership page
## Description

Now that we've published the CMS powered leadership page in prod, we can remove the old hard-coded page.

To-do-list:

1. Publish CMS /leadership/ page on dev & stage with placeholder test data to satisfy functional test runs.
2. Remove hard-coded leadership page and assets from bedrock, including `prefers_cms` URL decorators.
3. Update internal links to leadership page with `cms_url()` helper.

---

#### Success Criteria
- [x] Old leadership page is removed.

### PR DESCRIPTION
**Pull request by alexgibson**
## One-line summary

Removes old leadership page template

## Issue / Bugzilla link

#15135

## Testing

Remember to run `make preflight` to pull the latest dev DB.

- [ ] http://localhost:8000/en-US/about/leadership/ should still resolve to the CMS powered page

### CODE REVIEW

**Comment by alexgibson** [comment]
Successful test run: https://github.com/mozilla/bedrock/actions/runs/10939389570

**Review by reemhamz — COMMENTED** [review]
Hmm, this is the only thing I see on the page. It doesn't seem to resolve to the new CMS leadership page and made sure to run a `make preflight`. Is this something else I might be missing?

<kbd>Screenshot 2024-09-20 at 1 47 13 PM</kbd>

**Comment by alexgibson** [comment]
@reemhamz the localhost page should match what's on https://www-dev.allizom/en-Us/about/leadership/

See https://bedrock.readthedocs.io/en/latest/cms.html#i-ve-downloaded-a-fresh-db-and-the-images-are-missing for downloading images locally.

File: bedrock/mozorg/templates/mozorg/cms/about/leadership.html [hunk 1]
```diff
@@ -18,40 +26,46 @@
...
+          (fullpageurl(page), 'executive-team', page.title),
+        ] -%}
+        {% for section in page.leadership_sections %}
+          {{ nav_bar.append(('#' ~ section.value.id, section.value.id, section.value.title)) }}
+        {% endfor %}
+
+        {{ sidemenu_lists([nav_bar]) }}
+      </aside>
+
+      <main class="mzp-l-main">
```
Comment (stevejalim) [inline bedrock/mozorg/templates/mozorg/cms/about/leadership.html:42]: Does it matter that there are two `main` nodes here? one here and one at the top of the diff?

Comment (alexgibson) [reply bedrock/mozorg/templates/mozorg/cms/about/leadership.html:42]: Oh good call, you're right there should only be one. I'll update!

Comment (alexgibson) [reply bedrock/mozorg/templates/mozorg/cms/about/leadership.html]: @stevejalim updated, thanks

**Review by reemhamz — APPROVED** [review]
Giving this a front-end stamp of approval. 💯 r+
I'll leave this for you to merge in case

**Pull request merged**